iceoryx_dust/include/iceoryx_dust/internal/cxx/string_conversion.inl

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_DUST_STRING_CONVERSION_INL
+#define IOX_DUST_STRING_CONVERSION_INL
+
+#include "iceoryx_dust/cxx/string_conversion.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+template <uint64_t N>
+inline std::string FromImpl<string<N>, std::string>::fromImpl(const string<N>& value)
+{
+    return std::string(value.c_str(), value.size());
+}
+
+template <uint64_t N>
+inline string<N> FromImpl<std::string, string<N>>::fromImpl(const std::string& value) noexcept
+{
+    return string<N>(TruncateToCapacity, value.c_str(), value.size());
```

Comment:
I have to take counsel with my pillow regarding this. We introduced the `TruncateToCapacity` in order to make it visible to the user that not the whole string might be placed in the `cxx::string`. With this change it is hidden and the user might not be aware that something like this might happen. Since we use the logger in `string.inl`, we could at least add a debug log message when the truncation happens. @FerdinandSpitzschnueffler what do you think?

Reply:
@elBoberido There is already a message printed in that case:

https://github.com/eclipse-iceoryx/iceoryx/blob/36cae1d1c5ae80204e74be8d0ea82a43ed6c6723/iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl#L150-L151

Reply:
@elBoberido Yes, a log message (debug or warn) would be good. As @mossmaurice mentioned, there is already a message printed to the console if the string is truncated. It just needs to be replaced with `IOX_LOG` which is probably done in #1787(?) @mossmaurice What do you think of adding an additional Doxygen note to give the user a hint about the truncating behavior?

Reply:
@FerdinandSpitzschnueffler For me this is crystal clear with the first parameter `TruncateToCapacity_t`. The `IOX_LOG` change is indeed something for #1787

Reply:
@mossmaurice When I just look at `std_string_support.hpp`, I have no idea about `TruncateToCapacity_t`. A hint might be helpful, but up to you.

Reply:
@FerdinandSpitzschnueffler As this PR already has a approval, I would prefer to go ahead and merge. Could you approve? :pray: